### ISSUE TITLE
NameError: name 'podman_error' is not defined

### ISSUE THREAD

**Issue by GrabbenD** [polygamma/auv#1]
Are there any particular Python dependencies to run this project?

```console
$ pacman -S netavark crun podman
$ echo 'unqualified-search-registries = ["docker.io"]' | tee -a /etc/containers/registries.conf

# git clone https://github.com/polygamma/auv.git && cd auv/
(...)

# python3 src/main.py build_image
Traceback (most recent call last):
  File "/home/test/auv-main/src/main.py", line 66, in <module>
    class Calls:
  File "/home/test/auv-main/src/main.py", line 78, in Calls
    podmanRoot = loads(run_command(podmanInfo, True, valid_return_codes=(0,)).stdout)["store"]["graphRoot"].strip()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/test/auv-main/src/main.py", line 53, in run_command
    podman_error("'{}' could not be executed correctly:\n   {}".format(
    ^^^^^^^^^^^^
NameError: name 'podman_error' is not defined
```

Environment:
```console
# python --version
Python 3.11.4

# podman --version
podman version 4.6.0
```

I've also tested https://github.com/polygamma/auv/releases/tag/1.0 and got the same error

**Comment by polygamma** [comment]
Hey :)

What is the output of `sudo podman info --format=json` ?

**Comment by GrabbenD** [comment]
Here you go!
```
# sudo podman info --format=json
{
  "host": {
    "arch": "amd64",
    "buildahVersion": "1.31.0",
    "cgroupManager": "systemd",
    "cgroupVersion": "v2",
    "cgroupControllers": [
      "cpuset",
      "cpu",
      "io",
      "memory",
      "hugetlb",
      "pids",
      "rdma",
      "misc"
    ],
    "conmon": {
      "package": "/usr/bin/conmon is owned by conmon 1:2.1.7-1.1",
      "path": "/usr/bin/conmon",
      "version": "conmon version 2.1.7, commit: f633919178f6c8ee4fb41b848a056ec33f8d707d"
    },
    "cpus": 32,
    "cpuUtilization": {
      "userPercent": 0.64,
      "systemPercent": 0.23,
      "idlePercent": 99.13
    },
    "databaseBackend": "boltdb",
    "distribution": {
      "distribution": "arch",
      "version": "unknown"
    },
    "eventLogger": "journald",
    "freeLocks": 2047,
    "hostname": "unix",
    "idMappings": {
      "gidmap": null,
      "uidmap": null
    },
    "kernel": "6.4.7-arch1-2",
    "logDriver": "journald",
    "memFree": 52815495168,
    "memTotal": 67333218304,
    "networkBackend": "netavark",
    "networkBackendInfo": {
      "backend": "netavark",
      "version": "netavark 1.7.0",
      "package": "/usr/lib/podman/netavark is owned by netavark 1.7.0-1.1",
      "path": "/usr/lib/podman/netavark",
      "dns": {
        "package": "Unknown"
      }
    },
    "ociRuntime": {
      "name": "crun",
      "package": "/usr/bin/crun is owned by crun 1.8.5-1.1",
      "path": "/usr/bin/crun",
      "version": "crun version 1.8.5\ncommit: b6f80f766c9a89eb7b1440c0a70ab287434b17ed\nrundir: /run/user/0/crun\nspec: 1.0.0\n+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL"
    },
    "os": "linux",
    "remoteSocket": {
      "path": "/run/podman/podman.sock"
    },
    "serviceIsRemote": false,
    "security": {
      "apparmorEnabled": false,
      "capabilities": "CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT",
      "rootless": false,
      "seccompEnabled": true,
      "seccompProfilePath": "/etc/containers/seccomp.json",
      "selinuxEnabled": false
    },
    "slirp4netns": {
      "executable": "/usr/bin/slirp4netns",
      "package": "/usr/bin/slirp4netns is owned by slirp4netns 1.2.0-1.1",
      "version": "slirp4netns version 1.2.0\ncommit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383\nlibslirp: 4.7.0\nSLIRP_CONFIG_VERSION_MAX: 4\nlibseccomp: 2.5.4"
    },
    "pasta": {
      "executable": "",
      "package": "",
      "version": ""
    },
    "swapFree": 0,
    "swapTotal": 0,
    "uptime": "1h 19m 43.00s (Approximately 0.04 days)",
    "linkmode": "dynamic"
  },
  "store": {
    "configFile": "/etc/containers/storage.conf",
    "containerStore": {
      "number": 1,
      "paused": 0,
      "running": 0,
      "stopped": 1
    },
    "graphDriverName": "overlay",
    "graphOptions": {
      "overlay.mountopt": "nodev"
    },
    "graphRoot": "/var/lib/containers/storage",
    "graphRootAllocated": 399883579392,
    "graphRootUsed": 29123264512,
    "graphStatus": {
      "Backing Filesystem": "xfs",
      "Native Overlay Diff": "false",
      "Supports d_type": "true",
      "Using metacopy": "true"
    },
    "imageCopyTmpDir": "/var/tmp",
    "imageStore": {
      "number": 135
    },
    "runRoot": "/run/containers/storage",
    "volumePath": "/var/lib/containers/storage/volumes",
    "transientStore": false
  },
  "registries": {
    "search": [
  "docker.io"
]
  },
  "plugins": {
    "volume": [
      "local"
    ],
    "network": [
      "bridge",
      "macvlan",
      "ipvlan"
    ],
    "log": [
      "k8s-file",
      "none",
      "passthrough",
      "journald"
    ],
    "authorization": null
  },
  "version": {
    "APIVersion": "4.6.0",
    "Version": "4.6.0",
    "GoVersion": "go1.20.6",
    "GitCommit": "38e6fab9664c6e59b66e73523b307a56130316ae-dirty",
    "BuiltTime": "Wed Jul 26 03:23:31 2023",
    "Built": 1690334611,
    "OsArch": "linux/amd64",
    "Os": "linux"
  }
}
```

**Comment by polygamma** [comment]
Interesting. Sadly, I'm not at home but I'll look into it tomorrow.

However: You could try replacing the podman_error call with a normal print and execute it again.

And one more question: What is the return code of the podman info call you just executed?

**Comment by GrabbenD** [comment]
> Interesting. Sadly, I'm not at home but I'll look into it tomorrow.

I'm guessing it's because I'm using `crun` engine instead of `runc` (default), it has higher performance and probably different syntax.

> And one more question: What is the return code of the podman info call you just executed?

```
# echo $?
0
```

> However: You could try replacing the podman_error call with a normal print and execute it again.

I managed to get around it by extracting the podman commands from `main.py` to try the project out!

**Comment by polygamma** [comment]
Cool, I'll look into using crun instead of runc tomorrow, so that everything works with crun as well :)

If you encounter any more problems, please let me know :)

**Comment by polygamma** [comment]
And I'll rewrite the code making use of podman_error...

**Comment by polygamma** [comment]
Hey :)

So... First things first, see: https://github.com/polygamma/auv/commit/28d5076a46c686ac2c60784e9f8ccb23e0f7add4
"NameError: name 'podman_error' is not defined" should be gone.
Can you confirm?

> I'm guessing it's because I'm using crun engine instead of runc (default), it has higher performance and probably different syntax.

Why do you think that `runc` is the default? See: https://archlinux.org/packages/extra/x86_64/podman/
`crun` is a direct dependency, and `netavark` is a dependency of `containers-common` and that is another direct dependency of `podman`.
In other words: Both `crun` and `netavark` are defaults.

It boils down to: I can't reproduce the problem. But since the 'podman_error' definition should be fixed now, we can try to figure out what's going on, since you should get an error message now, without `src/main.py` crashing :)

**Comment by polygamma** [comment]
See: https://github.com/polygamma/auv/releases/tag/1.1

Let me know if there are still problems :)
